client/me/purchases/manage-purchase/index.jsx

```diff
@@ -562,6 +581,7 @@ class ManagePurchase extends Component {
 export default connect( ( state, props ) => {
 	const purchase = getByPurchaseId( state, props.purchaseId );
 	const siteId = purchase ? purchase.siteId : null;
+	const renewableSitePurchases = getRenewableSitePurchases( state, siteId );
```

Comment:
Since this creates and returns an array each time it is called, I _think_ that means that it will cause `purchases` to be considered changed on every render of `UpcomingRenewalsDialog`, which will trigger every `useEffect`, `useMemo`, and `useCallback` on every render. If I'm right, is that an issue? Maybe we should try to memoize this value somewhere? I also might be wrong, in which case I'd love to understand why.

Reply:
Good catch! I will wrap the selector using `createSelector` so it can be properly memoized.